core/shared/src/main/scala/scodec/bits/ByteVector.scala

```diff
@@ -777,6 +779,37 @@ sealed abstract class ByteVector extends BitwiseOperations[ByteVector, Long] wit
     bldr.toString
   }
 
+  /**
+    * Converts the contents of this vector to a base 58 string.
+    *
+    * @group conversions
+    */
+  final def toBase58: String = toBase58(Bases.Alphabets.Base58)
+
+  /**
+    * Converts the contents of this vector to a base 58 string using the specified alphabet.
+    *
+    * @group conversions
+    */
+  final def toBase58(alphabet: Bases.Alphabet): String = {
+    val bytes = toArray
+    val ZERO = BigInt(0)
+    val RADIX = BigInt(58L)
+    val ones = Vector.fill(takeWhile(_ == 0).length.toInt)(1)
+
+    @tailrec
+    def go(value: BigInt, chars: Vector[Char]): String = value match {
```

Comment:
Can `chars` be a `List[Char]`? Then we can prepend each char and also avoid the reverse call.